repo/manifest/committed_manifest_manager.go

```diff
@@ -132,6 +185,211 @@ func (m *committedManifestManager) writeEntriesLocked(ctx context.Context, entri
 	return map[content.ID]bool{contentID: true}, nil
 }
 
+// +checklocks:m.cmmu
+func (m *committedManifestManager) writeContentChunkLocked(
+	ctx context.Context,
+	entries []*inMemManifestEntry,
+	buf *gather.WriteBuffer,
+) error {
+	contents := contentSet{
+		Version: 1,
+	}
+
+	for _, e := range entries {
+		contents.Contents = append(
+			contents.Contents,
+			&manifestContent{
+				ID:      e.ID,
+				Content: e.Content,
+			},
+		)
+	}
+
+	contentID, err := m.writeContentLocked(
+		ctx,
+		IndirectContentPrefix,
+		contents,
+		buf,
+	)
+	if err != nil {
+		return errors.Wrapf(err, "writing manifest content chunk")
+	}
+
+	for _, e := range entries {
+		e.ContentID = contentID.String()
```

Comment:
tip: contentID will be the same, so it makes sense to create the string once and set it to the same value - they will be properly aliased.

Also when parsing this manifest, we can take advantage of ContentID interning.

Reply:
Will revisit interning for deserialization in a later PR as this one's getting a bit large